e3sm_diags/driver/utils/io.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import errno
+import os
+
+import xarray as xr
+
+from e3sm_diags.logger import custom_logger
+from e3sm_diags.parameter.core_parameter import CoreParameter
+
+logger = custom_logger(__name__)
+
+
+def _write_vars_to_netcdf(
+    parameter: CoreParameter,
+    test: xr.DataArray,
+    ref: xr.DataArray,
+    diff: xr.DataArray | None,
+):
+    """Saves the test, reference, and difference variables to netCDF files.
+
+    Parameters
+    ----------
+    parameter : CoreParameter
+        The parameter object used to configure the diagnostic runs for the
+        sets. The referenced attributes include `save_netcdf, `current_set`,
+        `var_id`, `ref_name`, and `output_file`, `results_dir`, and `case_id`.
+    test : xr.DataArray
+        The test variable.
+    ref : xr.DataArray
+        The reference variable.
+    diff : Optional[xr.DataArray]
+        The optional difference between the test and reference variables.
+
+    Notes
+    -----
+    Replaces `e3sm_diags.driver.utils.general.save_ncfiles()`.
+
+    # TODO: Do we need to consider the append option like in save_ncfiles()?
```

Comment:
It looks like this still needs to be addressed?

Reply:
This was a question for @chengzhuzhang. There is a section of logic for appending content to netCDF files with cdms2. Do we ever append data, or always write to a new file?

https://github.com/E3SM-Project/e3sm_diags/blob/c02d7015bbd1b211bdf3165d8412ace29ca23d5c/e3sm_diags/driver/utils/general.py#L312-L333

Reply:
The cdms2 implementation saves one variable per file. I think aggregate test, ref (if available) and diff (if available) will be slightly cleaner? I don't have strong preference here.

Reply:
I agree, it is cleaner to have a single netCDF that stores all variables. I'll update `_write_vars_to_netcdf()` to do this.

My question was if we ever appended data to existing files (old logic below)? I don't know if we ever do.
```python
 if Path(test_pth).is_file(): 
     cdms_arg = "a" 
```

Reply:
I don't recall we ever appended data to existing files.

Reply:
Updated `_write_vars_to_netcdf()` to output test, ref, and diff variables in a single `.nc` file in [`e544788` (#677)](https://github.com/E3SM-Project/e3sm_diags/pull/677/commits/e544788cb305e5577446a5d1bf06f6e958082a17), which also includes unit tests.